openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -775,13 +788,350 @@ function scale(IndicatorObj) {
+    if (weightChange > 0) {
+        mapboxGlLayerCreation(weightChange);
+    }
+
+    $('#webGlThematicSelection').change(function() {
+        mapboxGlLayerCreation();
+    });
+
+}
+
+function mapboxGlLayerCreation() {
+    // There are 4 cases for managing the state of the map:
+    // 1. The user has created a map for the first time
+    // 2. The user changes the map theme
+    // 3. The user changes the weights
+    // 4. The user changes the project
+
+    // Case 2:
+    // The data used for thematic map has not changed, so we
+    // keep the map source, destroy and recreate the map layers
+
+    // Case 3:
+    // The weight change will alter the underlying data, so we
+    // update the map style and source
+
+    // Case 3.1
+    // Only reload the map for the appropriate weight change level
+
+    // Case 4:
+    // A new project has been loaded into the application
+
+    var weightChange = 0;
+
+    if (arguments[0]) {
+        weightChange = arguments[0];
+    }
+
+    selectedIndicator = $('#webGlThematicSelection').val();
+
+    // Avoid unnecessarily redrawing the map.
+    // Only redraw the map when:
+        // 1) the level of the weight changes is lower than the level of the thematic map menu.
+            // Example 1: if the user is viewing the SVI thematic map, and then changes any theme, or
+            // primary indicator weight, the map must be refreshed.
+            // Example 2: If the user is viewing a theme thematic map (e.g. Economy), and then changes
+            // the SVI weight, the map should not be redrawn.
+        // 2) the thematic map menu has been changes
+        // 3) a new project is loaded into the map (case: weightChange = 0)
+    var selectedIndicatorLabel = $('#webGlThematicSelection option:selected').attr('class');
+
+    if (weightChange < parseInt(selectedIndicatorLabel || selectedIndicatorLabel !== '4')) {
+        if (weightChange !== 0) {
+            $('#absoluteSpinner').hide();
+            return;
+        }
+    }
+
+    // Find the values to create categorized color ramp
+    // First find the min and max vales
+    var minMaxArray = [];
+    for (var i = 0; i < mappingLayerAttributes.features.length; i++) {
+        var match = false;
+        for (var k in mappingLayerAttributes.features[i].properties) {
+            if (k == selectedIndicator) {
+                match = true;
+                minMaxArray.push(mappingLayerAttributes.features[i].properties[k]);
+                break
+            }
+            else {
+                match = false;
+            }
+        }
+        if (match === false) {
+            return;
+        }
+    }
+
+    // Focus map on layer
+    if (projectChange) {
+        map.fitBounds(mapboxBoundingBox);
+    }
+
+    var min = Math.min.apply(null, minMaxArray).toFixed(2);
+    var max = Math.max.apply(null, minMaxArray).toFixed(2);
+    min = (parseFloat(min) - 0.1);
+    // Round up the max
+    max = Math.ceil(max * 10) / 10;
```

Comment:
This values is used for the map filter scale breaks (L1035), which are being generated dynamically. So I need to round up. But if the max value is less then 1, for example 0.6555555, I do not want to round up to the nearest whole number (1), instead I want to round up to the first decimal place (0.7)